metpy/calc/indices.py

```diff
@@ -88,11 +88,18 @@ def mean_pressure_weighted(pressure, *args, **kwargs):
     heights = kwargs.pop('heights', None)
     bottom = kwargs.pop('bottom', None)
     depth = kwargs.pop('depth', None)
+    # Sort in decreasing pressure order
+    sort_inds = np.argsort(pressure[::-1])
```

Comment:
Not sure about having the pressure array be reversed going into argsort. Why not just argsort it as it stands? Should be fine either increasing or decreasing.

Reply:
The actual error gets thrown from get_layer, so get_layer itself might not be handling the reversed arrays correctly, and that might be where the actual problem is.  Unfortunately, since the error gets thrown at get_layer, I can't sort stuff afterward. As for reversing the array going into argsort, that was inherited from precipitable_water and can probably be changed.

Reply:
Should probably change it in precipitable_water too... 🐑

Reply:
[RAP sounding test.zip](https://github.com/Unidata/MetPy/files/1415823/RAP.sounding.test.zip)

Here's a somewhat cleaned-up version of the script where I found the problem. It shows up in the last two cells.